### PR TITLE
[hap2] Create new nagios plugin that use livestatus API

### PR DESCRIPTION
**Pull request by Mnakagawa**
livestatus API:
https://mathias-kettner.de/checkmk_livestatus.html

CentOS(from epel repository)
```
# yum install check-mk-livestatus
```

Ubuntu
```
# apt-get install check-mk-livestatus
```

Python library:
https://github.com/arthru/python-mk-livestatus
CentOS & Ubuntu
```
# pip install python-mk-livestatus
```

### CODE REVIEW

File: server/hap2/hatohol/hap2_nagios_livestatus.py [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+        query = self.__socket.hosts.columns("name", "alias")
+        result = query.call()
+        hosts = [{"hostId": host["name"], "hostName": host["alias"]} for host in result]
+        self.put_hosts(hosts)
+
+    def collect_host_groups_and_put(self):
+        query = self.__socket.hostgroups.columns("name", "alias")
+        result = query.call()
+        groups = \
+          [{"groupId": group["name"], "groupName": group["alias"]} for group in result]
```
Comment (cosmo0920) [inline server/hap2/hatohol/hap2_nagios_livestatus.py]: This lines does not use 4 multiples of spaces indentations.

File: server/hap2/hatohol/hap2_nagios_livestatus.py [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+                unix_timestamp = Utils.translate_hatohol_time_to_unix_time(self.__trigger_last_info)
+                query = query.filter("last_state_change >= %s" % unix_timestamp)
+                update_type = "UPDATED"
+
+        result = query.call()
+
+        triggers = []
+        for service in result:
+            hapi_status, hapi_severity = \
+              self.__parse_status_and_severity(service["state"])
```
Comment (cosmo0920) [inline server/hap2/hatohol/hap2_nagios_livestatus.py]: ditto.

File: server/hap2/hatohol/hap2_nagios_livestatus.py [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+        for event in result:
+            if not len(event["current_host_name"]): continue
+
+            hapi_event_type = self.EVENT_TYPE_MAP.get(event["state"])
+            if hapi_event_type is None:
+                log.warning("Unknown status: " + event["state"])
+                hapi_event_type = "UNKNOWN"
+
+            hapi_status, hapi_severity = \
+              self.__parse_status_and_severity(event["state"])
```
Comment (cosmo0920) [inline server/hap2/hatohol/hap2_nagios_livestatus.py]: ditto.

File: server/hap2/hatohol/hap2_nagios_livestatus.py [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+                                               "description",
+                                               "last_state_change",
+                                               "host_alias",
+                                               "host_name",
+                                               "state")
+
+        if host_ids is not None:
+            filter_condition = "host_name ~ "
+            for host_id in enumerate(host_ids):
+                if host_id[0] == 0: filter_condition += host_id[1]
```
Comment (cosmo0920) [inline server/hap2/hatohol/hap2_nagios_livestatus.py]: These two lines contains two statements in one line.
This condensed style is orthogonal against Python coding conventions.
We should do simple. One line should contain one statement.

**Comment by cosmo0920** [comment]
I'll merge this PR after hap2 tests finished in Travis.

**Comment by kz0817** [comment]
Thank you for the new plugin.

**Comment by Mnakagawa** [comment]
Oh... I'm sorry. I can not fixed some point yesterday.
@cosmo0920 san thank you for your fixing.
